### PR TITLE
add create secret for k8s v1.24+

### PR DESCRIPTION
**Pull request by guozheng-shen**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
Creating a ServiceAccount does not automatically generate Secret in v1.24+

**Which issue(s) this PR fixes**:
NONE

**Special notes for your reviewer**:
NONE
**Does this PR introduce a user-facing change?**:
NONE
```release-note

```

### CODE REVIEW

File: deploy/karmada-dashboard-secret.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+apiVersion: v1
+kind: Secret
+type: kubernetes.io/service-account-token
+metadata:
+  name: karmada-dashboard
+  namespace: karmada-system
+  annotations:
+    kubernetes.io/service-account.name: "karmada-dashboard"
```
Comment (warjiang) [inline deploy/karmada-dashboard-secret.yaml:8]: here need a new line, could you refine `deploy/karmada-dashboard-secret.yaml` file

Comment (guozheng-shen) [reply deploy/karmada-dashboard-secret.yaml:8]: ok

**Comment by warjiang** [comment]
@guozheng-shen could recheck dco?

**Pull request merged**